diff --git a/pyproject.toml b/pyproject.toml
index 56a64ba..59ac43d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -37,20 +37,20 @@ classifiers = [  # Optional
   "Programming Language :: Python :: 3.10",
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
   "Programming Language :: Python :: 3 :: Only",
   "Programming Language :: Fortran",
 ]
 
 dependencies = [
   "control",
-  "numpy<2.0",
+  "numpy",
   "matplotlib",
   "scipy",
   "pandas",
   "pyparsing",
   "pyYAML",
   "pyzmq",
-  "treon",
   "wisdem",
   "ruamel.yaml",
   "openfast_io",
@@ -66,7 +66,7 @@ dependencies = [
 # projects.
 [project.optional-dependencies] # Optional
 dev = ["cmake"]
-test = ["pytest"]
+test = ["pytest","treon"]
 
 # List URLs that are relevant to your project
 #
diff --git a/rosco/toolbox/__init__.py b/rosco/toolbox/__init__.py
index 511e363..6d0462b 100644
--- a/rosco/toolbox/__init__.py
+++ b/rosco/toolbox/__init__.py
@@ -1,7 +1,9 @@
 # -*- coding: utf-8 -*-
 """Top-level package for ROSCO Repo."""
-import pkg_resources
+import importlib
+import importlib.metadata
 
 __author__ = """Nikhar J. Abbas and Daniel S. Zalkind"""
 __email__ = 'daniel.zalkind@nrel.gov'
-__version__ = pkg_resources.get_distribution('rosco').version
+__version__ = importlib.metadata.version('rosco')
+
